src/Language/Haskell/LSP/Test/Decoding.hs

```diff
@@ -32,7 +32,7 @@ getNextMessage :: Handle -> IO B.ByteString
 getNextMessage h = do
   headers <- getHeaders h
   case read . init <$> lookup "Content-Length" headers of
-    Nothing   -> error "Couldn't read Content-Length header"
+    Nothing   -> throw NoContentLengthHeader
```

Comment:
This is not strictly needed using the shutdown and there are other uses of `error` in the module. However i think we could left as is.

Reply:
Yes I think this is actually a nice addition, seeing as this is really an exception with the server and not lsp-test itself